### ISSUE TITLE
expose rendering hooks for day headers

### ISSUE THREAD

**Issue by arshaw** [fullcalendar/fullcalendar#3078]
[requested by](https://github.com/fullcalendar/fullcalendar-scheduler/issues/142) @iamkoby and @mattdanna:

> Currently the only ability is to edit `slotLabelFormat`. 
> Let's say the format is "dd DD", I would love the option to edit css of that column but I can't separate classes of the "dd" and the "DD". The desired result is:
> ![screenshot 2016-02-24 12 03 02](https://cloud.githubusercontent.com/assets/2076259/13281819/94c4d458-daee-11e5-9edb-05f3e741ec35.png)
> 
> I am now patching this with editing the html in the viewRender function:
> 
> ```
> viewRender: function(view, element){
>     element.find('.fc-head .fc-time-area th.fc-widget-header .fc-cell-text').each(function(i, col){
>         var content = $(col).text().trim().split(',');
>         $(col).html('<span class="day-name">' + content[0] + '</span><span class="day-num">' + content[1] + '</span>');
>     });
> },
> ```

**Comment by arshaw** [comment]
same issue: https://github.com/fullcalendar/fullcalendar/issues/3331

**Comment by Tatikyan** [comment]
Hello Arshaw ,
I want to change slotDuration to 1 second as  I drop on timeline videos , but browser is lagging , not responding , how can I fix this ? .
Also is there any way that events overlap and don't get bottom second event other concatenate each other ?
Thanks a lot.

**Comment by arshaw** [comment]
I'm confused whether the request is for `slotLabelFormat` or `columnFormat` but I supposed we'd want to develop the same API improvement for each

**Comment by caseyjhol** [comment]
@arshaw You mentioned the following API in another pull request:

```
columnHead: true/false
columnHeadFormat: date format string
columnHeadText: function
columnHeadHtml: function
```

Can you elaborate a bit on how you want that to work? I'm not entirely sure how you want columnHeadFormat, columnHeadText, and columnHeadHtml to be related.

**Comment by caseyjhol** [comment]
@arshaw Mainly, I'm curious, if a user has defined columnHeadFormat, do you want to then pass the resulting string into the columnHeadText and columnHeadHtml functions? Then, does the user need to decide between using columnHeadText or columnHeadHtml, or can they be used in conjunction with each other? What do you propose as the best method for passing through the result of columnHeadFormat into the columnHeadHtml function? For example, if the result is after formatting is `Thu 8/3`, do you want to just pass through that whole string as the only argument, and leave it up to the developer to decide how to break it up to put the day name and the date (e.g. `string.split(' ')`) wherever they want in the HTML? Or would you rather pass through `ddd M/D` before it is formatted, and leave it up to the developer to call moment.format?

**Comment by arshaw** [comment]
@caseyjhol , sorry for the delay. does this answer your questions?

```js
columnHeadFormat: 'asdf'

// if columnHeadText is defined, columnHeadFormat will be ignored
columnHeadText: function(mom) {
   return 'the date & time is ' + mom.format('asdf'); // will be automatically escaped
}

// if columnHeadHtml is defined, columnHeadText and columnHeadFormat will be ignored
columnHeadHtml: function(mom) {
    return '<marquee>' + mom.format('asdf') + '</marquee>'; // won't be escaped
}
```

**Comment by almgwary** [comment]
thanks @arshaw  [your solution works  ](https://github.com/fullcalendar/fullcalendar/issues/3078#issue-138849808) 
i just changed selector to match mine 


```
element.find('th.fc-day-header span').each(function(i, col){
          var content = $(col).text().trim().split(',');
          $(col).html('<span class="day-name">' + content[0] + '</span><span class="day-num">' + content[1] + '</span>');
        });

```

**Comment by arshaw** [comment]
released in v3.8.0!
https://github.com/fullcalendar/fullcalendar/releases/tag/v3.8.0

**Comment by arshaw** [comment]
i still need to write docs though
